content/web-api-rest/reading/web-api/_index.md

```diff
@@ -0,0 +1,264 @@
+
+{{% notice green "Tip" "rocket" %}}
+[XML](https://developer.mozilla.org/en-US/docs/Web/XML/XML_introduction) is another popular data format. It is now used less commonly than 
+JSON for web API-to-client communications.
+{{% /notice %}}
+
+In the next section, we will discuss exactly how a client application makes a request and how a web API responds.
+
+### HTTP as The Language of Web APIs
+
+HTTP is the protocol used for communication between a web API and a client application. Web APIs communicate over a network. The most common protocol of the internet is HTTP, so it comes as no surprise that HTTP is the language of web APIs. Similarly, our MVC applications use HTTP as the protocol for an end-user to access the application. HTTP also facilitates the communication between a client application and a web API.
+
+{{% notice green "Tip" "rocket" %}}
+We will refer to web APIs as APIs going forward, since the web prefix is implied.
+{{% /notice %}}
+
+Here's a refresher on [the basics of HTTP](https://education.launchcode.org/intro-to-professional-web-dev/chapters/http/index.html):
+
+- Is a stateless request/response protocol.
+- Requests and responses may include HTTP bodies.
+- Responses always contain a three digit HTTP status code.
+- Requests and responses always include HTTP headers.
+
+
+We call HTTP a *stateless protocol*. State can be a complex concept that refers to a number of things. We'll explore some aspects of it in more depth on the next page. 
+
+In the context of HTTP, think of **state** as information about application data that is transferred via HTTP bodies, HTTP status codes, and HTTP headers.
+
+#### Bodies
+
+An HTTP body can contain a large number of different media types, 
+known as [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). A MIME type is associated with the HTTP header `Content-Type`. This header instructs the recipient of the HTTP request/response on what MIME type the HTTP body contains. We've seen a `Content-Type:text/html` HTTP header before. Here's a simple example of it:
+
+```html {linenos=table}
+<!DOCTYPE html>
+<html>
+   <head>
+      <title>My Web Page</title>
+      content
+   </head>
+   <body>
+      content
+   </body>
+</html>
+```
+
+This is the header for HTML documents and is used throughout the web. APIs send representations of data in the format of JSON requiring the header `Content-Type` to be `application/json`. This allows us to pass the state of the data as the HTTP body.
+
+```json {linenos=table}
+{
+   "title": "An Astronaut's Guide to Life on Earth",
+   "author": "Chris Hadfield",
+   "ISBN": 9780316253017,
+   "year_published": 2013,
+   "subject": ["Hadfield, Chris", "Astronauts", "Biography"],
+   "available": true
+}
+```
+
+The HTTP body may include JSON that represents the data being passed between an API and client application. Remember, not all requests/responses include HTTP bodies.
```

Comment:
```suggestion
An HTTP body can contain a large number of different media types, 
known as [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). A MIME type is associated with the HTTP header `Content-Type`. This header instructs the recipient of the HTTP request/response on what MIME type the HTTP body contains. We've seen a `Content-Type:text/html` HTTP header before. 

{{% notice green "Example" "rocket" %}}
```html {linenos=table}
<!DOCTYPE html>
<html>
   <head>
      <title>My Web Page</title>
      content
   </head>
   <body>
      content
   </body>
</html>
```

This is the header for HTML documents and is used throughout the web. APIs send representations of data in the format of JSON requiring the header `Content-Type` to be `application/json`. This allows us to pass the state of the data as the HTTP body.

```json {linenos=table}
{
   "title": "An Astronaut's Guide to Life on Earth",
   "author": "Chris Hadfield",
   "ISBN": 9780316253017,
   "year_published": 2013,
   "subject": ["Hadfield, Chris", "Astronauts", "Biography"],
   "available": true
}
```

The HTTP body may include JSON that represents the data being passed between an API and client application. Remember, not all requests/responses include HTTP bodies.
{{% notice %}}
```